File: GFG/Next Greater Element.cpp
    vector<long long> nextLargerElement(vector<long long> arr, int n){
        vector<long long>ans(n);
        stack<long long>s;
        
        for(int i=n-1;i>=0;i--)
        {
            while(!s.empty() && arr[i] >= s.top())
            {
                s.pop();
            }
            ans[i] = (s.empty())?-1:s.top();
            s.push(arr[i]);
        }
        return ans;
    }

File: GFG/Next Greater Element/Next Greater Element.java
class Solution{
    public static long[] nextLargerElement(long[] arr, int n) { 
        // Your code here
        long[]ans=new long[n];
        Stack<Long> st=new Stack<>();
        for(int i=n-1;i>=0;i--){
            while(!st.isEmpty() && st.peek()<arr[i]){
                st.pop();
            }
            if(st.isEmpty()){
                ans[i]=-1;
            }else{
                ans[i]=st.peek();
            }
            st.push(arr[i]);
        }
        return ans;
    } 
}
